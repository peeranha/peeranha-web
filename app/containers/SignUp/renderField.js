import React from 'react';
import PropTypes from 'prop-types';

function renderField({
  input,
  label,
  type,
  disabled,
  readOnly,
  translations,
  meta: { touched, error, warning },
}) {
  return (
    <div>
      <h6>{label[0]}</h6>
      <input
        {...input}
        disabled={disabled}
        placeholder={label[0]}
        type={type}
        readOnly={readOnly}
        value={readOnly ? label[1] : undefined}
        className="form-control"
      />
      <h6 className="text-danger">
        {touched &&
          ((error && <span>{translations[error]}</span>) ||
            (warning && <span>{translations[warning]}</span>))}
      </h6>
    </div>
  );
}

renderField.propTypes = {
  input: PropTypes.object,
  label: PropTypes.array,
  type: PropTypes.string,
  readOnly: PropTypes.bool,
  disabled: PropTypes.bool,
  meta: PropTypes.object,
  translations: PropTypes.object,
};

export default renderField;
