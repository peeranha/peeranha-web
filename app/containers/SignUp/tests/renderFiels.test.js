import renderField from '../renderField';

describe('renderField', () => {
  const props = {
    input: {},
    label: [],
    type: 'type',
    disabled: false,
    readOnly: false,
    translations: [],
    meta: {
      touched: false,
      error: 'error',
      warning: 'warning',
    },
  };

  it('case1: snapshot test, @readOnly is false', () => {
    props.readOnly = false;
    expect(renderField(props)).toMatchSnapshot();
  });

  it('case2: snapshot test, @readOnly is true', () => {
    props.readOnly = true;
    expect(renderField(props)).toMatchSnapshot();
  });

  it('case3: snapshot test, @touched is true', () => {
    props.meta.touched = true;
    expect(renderField(props)).toMatchSnapshot();
  });

  it('case4: snapshot test, @touched is false', () => {
    props.meta.touched = false;
    expect(renderField(props)).toMatchSnapshot();
  });

  it('case5: snapshot test, @error is true', () => {
    props.meta.touched = true;
    props.meta.error = true;
    expect(renderField(props)).toMatchSnapshot();
  });

  it('case6: snapshot test, @error is false', () => {
    props.meta.touched = true;
    props.meta.error = false;
    expect(renderField(props)).toMatchSnapshot();
  });

  it('case7: snapshot test, @warning is true', () => {
    props.meta.touched = true;
    props.meta.warning = true;
    expect(renderField(props)).toMatchSnapshot();
  });

  it('case8: snapshot test, @warning is false', () => {
    props.meta.touched = true;
    props.meta.warning = false;
    expect(renderField(props)).toMatchSnapshot();
  });
});
