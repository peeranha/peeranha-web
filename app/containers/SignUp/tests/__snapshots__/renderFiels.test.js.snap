// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderField case1: snapshot test, @readOnly is false 1`] = `
<div>
  <h6 />
  <input
    className="form-control"
    disabled={false}
    readOnly={false}
    type="type"
  />
  <h6
    className="text-danger"
  />
</div>
`;

exports[`renderField case2: snapshot test, @readOnly is true 1`] = `
<div>
  <h6 />
  <input
    className="form-control"
    disabled={false}
    readOnly={true}
    type="type"
  />
  <h6
    className="text-danger"
  />
</div>
`;

exports[`renderField case3: snapshot test, @touched is true 1`] = `
<div>
  <h6 />
  <input
    className="form-control"
    disabled={false}
    readOnly={true}
    type="type"
  />
  <h6
    className="text-danger"
  >
    <span />
  </h6>
</div>
`;

exports[`renderField case4: snapshot test, @touched is false 1`] = `
<div>
  <h6 />
  <input
    className="form-control"
    disabled={false}
    readOnly={true}
    type="type"
  />
  <h6
    className="text-danger"
  />
</div>
`;

exports[`renderField case5: snapshot test, @error is true 1`] = `
<div>
  <h6 />
  <input
    className="form-control"
    disabled={false}
    readOnly={true}
    type="type"
  />
  <h6
    className="text-danger"
  >
    <span />
  </h6>
</div>
`;

exports[`renderField case6: snapshot test, @error is false 1`] = `
<div>
  <h6 />
  <input
    className="form-control"
    disabled={false}
    readOnly={true}
    type="type"
  />
  <h6
    className="text-danger"
  >
    <span />
  </h6>
</div>
`;

exports[`renderField case7: snapshot test, @warning is true 1`] = `
<div>
  <h6 />
  <input
    className="form-control"
    disabled={false}
    readOnly={true}
    type="type"
  />
  <h6
    className="text-danger"
  >
    <span />
  </h6>
</div>
`;

exports[`renderField case8: snapshot test, @warning is false 1`] = `
<div>
  <h6 />
  <input
    className="form-control"
    disabled={false}
    readOnly={true}
    type="type"
  />
  <h6
    className="text-danger"
  />
</div>
`;
